suite-native/module-device-settings/src/components/DeviceAuthenticityCard.tsx

```diff
@@ -48,13 +58,22 @@ export const DeviceAuthenticityCard = () => {
 
         const { success, payload } = result.payload;
         if (success) {
-            navigation.navigate(DeviceAuthenticityStackRoutes.AuthenticitySummary, {
-                checkResult: payload.valid ? 'successful' : 'compromised',
-            });
+            const checkResult = payload.valid ? 'successful' : 'compromised';
+            navigation.navigate(DeviceAuthenticityStackRoutes.AuthenticitySummary, { checkResult });
+            reportCheckResult(checkResult);
         } else if (payload.code === 'Failure_ActionCancelled') {
             navigation.goBack();
+            reportCheckResult('cancelled');
+        } else if (payload.code === 'Method_Interrupted') {
+            // navigation.goBack() already called by the X icon button
+            reportCheckResult('cancelled');
+        } else {
```

Comment:
Are we sure that navigation.goBack is called in this case as well? Are we sure that there is no scenario where user will get stuck on this page?